src/rules/store/prefer-action-creator.ts

```diff
@@ -0,0 +1,40 @@
+import type { TSESTree } from '@typescript-eslint/experimental-utils'
+import { ESLintUtils } from '@typescript-eslint/experimental-utils'
+import path from 'path'
+import { docsUrl } from '../../utils'
+
+export const messageId = 'preferActionCreator'
+export type MessageIds = typeof messageId
+
+type Options = []
+
+export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
+  name: path.parse(__filename).name,
+  meta: {
+    type: 'problem',
+    docs: {
+      category: 'Best Practices',
+      description:
+        'Using an `action creator` is preferred over `Action class`.',
+      recommended: 'warn',
+    },
+    schema: [],
+    messages: {
+      [messageId]:
+        'Using `Action class` is not preferred, use an `action creator` instead.',
+    },
+  },
+  defaultOptions: [],
+  create: (context) => {
+    return {
+      [`ClassDeclaration:has(TSClassImplements:matches([expression.name='Action'], [expression.property.name='Action']))`](
```

Comment:
Do you see any problem looking for `Action` like this? I'm asking because I have the impression that the `Action` is a very trivial naming and may not always be the one from `@ngrx/store`... on the other hand, we have many other rules where we look for various methods/functions no matter where it comes from, that I don't think having more of this would be a problem.

Reply:
I think this is fine.
For extra safety we could also verify that the class has a `type` property?